app/src/main/java/com/gathering/android/event/home/EventListViewModel.kt

```diff
@@ -66,9 +79,32 @@ class EventListViewModel @Inject constructor() : ViewModel() {
 
             }
         }
+
+        if (sortType == SortType.SORT_BY_DATE) {
+            filteredEventList = filteredEventList.sortedBy { it.date }
+        } else {
+            filteredEventList = filteredEventList.sortedWith(object : kotlin.Comparator<Event> {
+                override fun compare(o1: Event?, o2: Event?): Int {
+                    val distA = o1?.location?.distanceTo(getCurrentLocation())
+                    val distB = o2?.location?.distanceTo(getCurrentLocation())
+                    if (distA == null && distB == null) {
+                        return 0
+                    }
+                    if (distA == null) {
+                        return -1
+                    }
+                    if (distB == null) {
+                        return 1
+                    }
```

Comment:
the code should work just fine without these 3 if conditions. for that look into Float class comparable implementation methods to see if null cases has been handled. if yes we can remove these.